### PR TITLE
Google Summer of Code 2014

### PR DESCRIPTION
**Pull request by dLobatog**
This commit contains all old proposals for GSoC in Redmine. New
proposals should come in through PR instead of editing the Redmine wiki
page.

### CODE REVIEW

File: gsoc.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+layout: contribute
+title: Google Summer of Code 2014
```
Comment (domcleal) [inline gsoc.md]: Maybe remove the date if this is permanent.

File: gsoc.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+---
+## Mentors
+
+Please see [GSoC mentoring](http://en.flossmanuals.net/GSoCMentoring/) and [Federico's GSoC howto](https://people.gnome.org/~federico/docs/summer-of-code-mentoring-howto/) for information on mentoring.
+
+## Project Suggestions
+
+Please list project selections below. Send a pull request with the content of your proposal to github.com/theforeman.org. See [Dave Neary's post](http://www.outercurve.org/Blogs/EntryId/45/Making-the-most-of-Google-Summer-of-Code-Dave-Neary-guest-blogger) for
```
Comment (domcleal) [inline gsoc.md]: github.com/theforeman/theforeman.org (maybe just link the text to it)

File: gsoc.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+suggestions on how to propose ideas. They should be useful to the community, and achievable by a student in the time allotted.
+
+Please use the following format:
```
Comment (domcleal) [inline gsoc.md]: Perhaps add this as a comment, since it's not a wiki.

File: gsoc.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+* *Status*: Not started
+* *Summary of idea*: Integration of Puppet Forge into Foreman
+* *Knowledge prerequisite*: Familiarity with Ruby, configuration management. Experience with infrastructure a plus
+* *Skill level*: Medium
+* *Contacts*:
+* *Mentor(s)*: Daniel Lobato Garcia
+* *Notes*: In a similar way to what Foreman is doing now to pick up provisioning templates from the community, I believe we could take advantage of the Puppet Forge to allow users to use modules from there. Importing these modules into a new model, and importing the puppet classes into Foreman should be achievable by a motivated student with help from the community. This will allow Foreman users to configure their hosts without having to worry about keeping repositories for modules already in the Puppet Forge.
+
+GregSutcliffe: This is pretty close to what we get from Katello (via Pulp) so I'd be wary of duplicating effort here. In addition, we've historically always maintained the position of "Foreman doesn't modify your manfiests, it just parses them for info" and I'd be wary of blurring that line - we get enough questions that should really be in #puppet already :)
```
Comment (domcleal) [inline gsoc.md]: We probably shouldn't add comments in here?

File: gsoc.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+### Compute resource (virtualization) benchmarks
+
+* *Status*: Not started
+* *Summary of idea*: Benchmark compute resources, graph them and use this information in Foreman.
+* *Knowledge prerequisite*: Familiarity with Ruby, configuration management. Experience with infrastructure a plus.
+* *Skill level*: Medium
+* *Contacts*:
+* *Mentor(s)*: Daniel Lobato Garcia
+* *Notes*: Compute resources could be expensive. Whether you are managing your own virtualization resources with Ovirt, or you're using EC2 in Amazon, currently Foreman does not offer you a way to check what are the utilization stats for this. We should offer a way for users to do this and graph this information. This information can be also used to scale up and down your instances, using some policies like keeping track of energy savings, morning-night usage, etc..
+* *Internal Note: It would be useful to track compute resources usage per host group.
```
Comment (lzap) [inline gsoc.md:45]: I have a small prototype for this already. Not worth sharing yet, need to work on it more. :-)

Comment (dLobatog) [inline gsoc.md:45]: Maybe its worth sending a PR after this one if you know of some features a student could help you with?

**Comment by domcleal** [comment]
Thanks @eLobato!

**Pull request merged**